agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/AzureFunctionsLogProcessor.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.monitor.opentelemetry.exporter.implementation.AiSemanticAttributes;
+import com.microsoft.applicationinsights.agent.bootstrap.AzureFunctionsCustomDimensions;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.logs.LogProcessor;
+import io.opentelemetry.sdk.logs.ReadWriteLogRecord;
+
+public class AzureFunctionsLogProcessor implements LogProcessor {
+
+  private static final ClientLogger logger = new ClientLogger(AzureFunctionsLogProcessor.class);
+
+  @Override
+  public void onEmit(ReadWriteLogRecord logRecord) {
+    AzureFunctionsCustomDimensions customDimensions =
+        AzureFunctionsCustomDimensions.fromContext(Context.current());
```

Comment:
```suggestion
    AzureFunctionsCustomDimensions customDimensions =
        AzureFunctionsCustomDimensions.fromContext(Context.current());
    if (customDimensions == null) {
      // could log warning since this is unexpected
      return;
    }
```